### PR TITLE
feat(andsvc-protocol): Enable cleanup of service's temporary files after lifetime exceeded

### PR DESCRIPTION
**Pull request by ThanyaLeif**
#### Details

Before this PR, the temporary files created by the service were deleted at the beginning and at the closing of the application.
This PR schedules the deletion of these files 5 minutes after their creation.

##### Motivation

Avoid problems with the internal storage of the user's device, fix issue #129 

##### Context

We use a Working Manager, which currently schedules the elimination of all temporary files created by the service, with a life time greater than 5 min.

In case the cache directory does not exist or the files cannot be removed, the Working Manager simply ignores the problem.

The corresponding unit tests were also added to verify the behavior described above.

#### Pull request checklist

- [x] Addresses an existing issue: #129
- [x] Added/updated relevant unit test(s)
- [x] Ran `./gradlew fastpass` from `AccessibilityInsightsForAndroidService`
- [x] PR title _AND_ final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`).

### CODE REVIEW

File: AccessibilityInsightsForAndroidService/.idea/deploymentTargetDropDown.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (madalynrose) [inline AccessibilityInsightsForAndroidService/.idea/deploymentTargetDropDown.xml:1]: we should add this file to `.gitignore` and exclude it from being added

File: AccessibilityInsightsForAndroidService/.idea/runConfigurations.xml [hunk 2]
```diff
@@ -1,10 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (madalynrose) [inline AccessibilityInsightsForAndroidService/.idea/runConfigurations.xml:1]: do you remember why this file was removed? we might want to `.gitignore` this file as well if it was deleted as part of debugging something else.

**Review by madalynrose — CHANGES_REQUESTED** [review]
The code itself looks great! A couple of these files can be ignored in `.gitignore` and then I think it's good to go!

File: AccessibilityInsightsForAndroidService/app/build.gradle [hunk 3]
```diff
@@ -102,6 +102,7 @@ dependencies {
     implementation 'com.google.guava:guava:30.1.1-android'
 
     // Dev dependencies (not redistributed)
+    implementation("androidx.work:work-runtime:2.5.0")
```
[the remark this answers is not in the document]
Comment (ThanyaLeif) [reply AccessibilityInsightsForAndroidService/app/build.gradle]: I changed this in my last commit.

File: AccessibilityInsightsForAndroidService/.idea/deploymentTargetDropDown.xml (hunk 1 above)
Comment (ThanyaLeif) [reply AccessibilityInsightsForAndroidService/.idea/deploymentTargetDropDown.xml:1]: This file also seems to exclusively affect the emulator's settings. I also added to `.gitignore` in my last commit.

File: AccessibilityInsightsForAndroidService/.gitignore [hunk 4]
```diff
@@ -10,6 +10,8 @@
 /.idea/modules.xml
 /.idea/navEditor.xml
 /.idea/workspace.xml
+.idea/deploymentTargetDropDown.xml
```
Comment (madalynrose) [inline AccessibilityInsightsForAndroidService/.gitignore:13]: should this have a `/` before it like the others?

File: AccessibilityInsightsForAndroidService/.idea/deploymentTargetDropDown.xml (hunk 1 above)
Comment (madalynrose) [reply AccessibilityInsightsForAndroidService/.idea/deploymentTargetDropDown.xml:1]: We should also delete it and push the deleted file so it isn't tracked at all

**Review by madalynrose — CHANGES_REQUESTED** [review]
A couple more changes and I think it's good to go!

**Comment by dbjorge** [comment]
It would be good if you could update the title of the PR to something more descriptive; it looks like you might have missed updating a placeholder (it still says `summary of your changes`)

File: AccessibilityInsightsForAndroidService/.gitignore (hunk 4 above)
Comment (ThanyaLeif) [reply AccessibilityInsightsForAndroidService/.gitignore]: You're right
I corrected it in my last commit

File: AccessibilityInsightsForAndroidService/.idea/deploymentTargetDropDown.xml (hunk 1 above)
Comment (ThanyaLeif) [reply AccessibilityInsightsForAndroidService/.idea/deploymentTargetDropDown.xml]: Done!

Comment (ThanyaLeif) [reply AccessibilityInsightsForAndroidService/.idea/deploymentTargetDropDown.xml]: Done!
I excluded this file in my last commit.

**Review by madalynrose — APPROVED** [review]

**Pull request merged**